docs/guides/field-extensions.md

```diff
@@ -0,0 +1,172 @@
+
+<Note>
+
+The following examples only cover sync execution. To use extensions in async contexts,
+please have a look at [Async Extensions and Resolvers](#Async Extensions and Resolvers)
+
+</Note>
+
+```python
+import strawberry
+from strawberry.extensions import FieldExtension
+
+
+class UpperCaseExtension(FieldExtension):
+    def resolve(self, next_: Callable[..., Any], source: Any, info: Info, **kwargs):
+        result = next_(source, info, **kwargs)
+        return str(result).upper()
+
+
+@strawberry.type
+class Query:
+    @strawberry.field(extensions=[UpperCaseExtension()])
+    def string(self) -> str:
+        return "This is a test!!"
+```
+
+In this example, the `UpperCaseExtension` wraps the resolver of the `string`
+field (`next`) and modifies the resulting string to be uppercase.
+The extension will be called instead of the resolver and receives
+the resolver function as the `next` argument. Therefore, it is important
+to not modify any arguments that are passed to `next` in an incompatible way.
+
+```graphql+response
+query {
+    string
+}
+---
+{
+  "string": "THIS IS A TEST!!"
+}
+```
+
+## Modifying the field
+
+<Warning>
+The `StrawberryField` API is not stable and might change in the future without warning.
+</Warning>
+
+In some cases, the extended field needs to be compatible with the added extension.
+`FieldExtension` provides an `apply(field: StrawberryField)` method that can be
+overriden to modify the field. It is called during _Schema Conversion_.
+In the following example, we use `apply` to add a directive to the field:
+
+```python
+import time
+import strawberry
+from strawberry.extensions import FieldExtension
+from strawberry.schema_directive import Location
+from strawberry.field import StrawberryField
+
+
+@strawberry.schema_directive(locations=[Location.FIELD_DEFINITION])
+class Cached:
+    time: int = 100
+
+
+class CachingExtension(FieldExtension):
+    def __init__(self, caching_time=100):
+        self.caching_time = caching_time
+        self.last_cached = 0.0
+        self.cached_result = None
+
+    def apply(self, field: StrawberryField) -> None:
+        field.directives.append(Cached(time=self.caching_time))
+
+    def resolve(
+        self, next_: Callable[..., Any], source: Any, info: Info, **kwargs
+    ) -> Any:
+        current_time = time.time()
+        if self.last_cached + self.caching_time > current_time:
+            return self.cached_result
+        self.cached_result = next_(source, info, **kwargs)
+        return self.cached_result
+```
+
+```python+schema
+@strawberry.type
+class Client:
+
+    @strawberry.field(extensions=[CachingExtensions(caching_time=200)])
+    def analyzed_hours(self, info) -> int:
+        return do_expensive_computation()
+---
+type Client {
+    analyzedHours: Int! @Cached(time=200)
+}
+```
+
+## Combining multiple field extensions
+
+When chaining multiple field extensions, the outermost extension is called first,
```

Comment:
do you think we could replace `outermost` with last?

or could we make it so that the order is the same as the one you pass? I think the current behaviour might be confusing 😊

Reply:
let's use mermaid, it should work on github too: https://mermaid.js.org

Reply:
Something like this?: https://mermaid.live/edit#pako:eNqNkVFLwzAQx79KOF8UWthzB0Kr86mipN2DLDJCe3XFNClpqi1j391rbJkOBPOU3O_3vwvJEQpTIkTwZmV7YClfC81odW5UyLI85jmraqWiqwe_VquAmDXveK5cRJ6e_5uY3B17XQ40aycAByx6VxvNCqMdDk7AbMREt22L9k52uBkc6o40wnO7hHhqPv_knDjHzqgPtOeqnxuGAuw32lc1qvL6RkAY3sazFE_C4wvjm2yb5h753E-s6a77OZcsV5pAMzLq3Sv3q2VymVmenvtZI-PnTLKGABq0jaxL-qvjJApwB2xQQETbEivpZaFPpMremWzUBUTO9hhA35bS4X0t6ZcbiCqpOjx9AUu0mVA

Reply:
yup, maybe vertical, as it might be easier to read on mobile 😊

Reply:
Does the strawberry website Markdown support html with media queries? I think it's much easier to read horizontally on desktop, but much better vertically on mobile so best to have both, what do you think? 👀

Reply:
Ok I just realized that doesn't work because the content inside those tags (the diagram in this case) won't be parsed by markdown then.

Reply:
@erikwrede we could make a custom component for that if it helps

or add a directive, like

````
```mermaid (mobile:vertical)
````

if it is easy to change the setting 😊

Reply:
Ok it actually looks just fine on the github preview. The strawberry preview doesn't seem to update though.

Reply:
it's probably because we don't support mermaid yet on the site :D I'll see if I can add support to the current design in the next couple of days 😊

Reply:
We could also add the graph later when the new website is ready